File: Java/Easy/1323.java
class Solution {
    public int maximum69Number(int num) {
        // hold the position of the most left 6 to change to a 9
        int pos = -1;
        int count = 0;
        int temp = num;
        while (temp > 0) {
            if (temp % 10 == 6) {
                pos = count;
            }
            count++;
            temp /= 10;
        }

        num = pos != -1 ? num + (int) Math.pow(10, pos) * 3 : num;
        return num;
    }
}

File: C/Easy/1323.c
int maximum69Number(int num)
{
    // always change the first 6
    int temp = num;
    int count = 0; // figure out which index is the 6
    int power = -1;
    while (temp > 0)
    {
        if (temp % 10 == 6)
        {
            power = count;
        }
        temp /= 10;
        count++;
    }
    int answer = num;
    if (power != -1)
    {
        answer = num - 6 * pow(10, power) + 9 * pow(10, power);
    }

    return answer;
}